cmd/livekit-cli/main.go

```diff
@@ -1,97 +1,11 @@
-// Copyright 2023 LiveKit, Inc.
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
 package main
 
 import (
 	"fmt"
 	"os"
-
-	"github.com/urfave/cli/v2"
-
-	livekitcli "github.com/livekit/livekit-cli"
-	"github.com/livekit/protocol/logger"
-	lksdk "github.com/livekit/server-sdk-go/v2"
 )
 
 func main() {
-	app := &cli.App{
-		Name:                 "livekit-cli",
-		Usage:                "CLI client to LiveKit",
-		Version:              livekitcli.Version,
-		EnableBashCompletion: true,
-		Suggest:              true,
-		Flags: []cli.Flag{
-			&cli.BoolFlag{
-				Name: "verbose",
-			},
-		},
-		Commands: []*cli.Command{
-			{
-				Name:   "generate-fish-completion",
-				Action: generateFishCompletion,
-				Hidden: true,
-				Flags: []cli.Flag{
-					&cli.StringFlag{
-						Name:    "out",
-						Aliases: []string{"o"},
-					},
-				},
-			},
-		},
-		Before: func(c *cli.Context) error {
-			logConfig := &logger.Config{
-				Level: "info",
-			}
-			if c.Bool("verbose") {
-				logConfig.Level = "debug"
-			}
-			logger.InitFromConfig(logConfig, "livekit-cli")
-			lksdk.SetLogger(logger.GetLogger())
-
-			return nil
-		},
-	}
-
-	app.Commands = append(app.Commands, TokenCommands...)
-	app.Commands = append(app.Commands, RoomCommands...)
-	app.Commands = append(app.Commands, JoinCommands...)
-	app.Commands = append(app.Commands, EgressCommands...)
-	app.Commands = append(app.Commands, IngressCommands...)
-	app.Commands = append(app.Commands, LoadTestCommands...)
-	app.Commands = append(app.Commands, ProjectCommands...)
-	app.Commands = append(app.Commands, SIPCommands...)
-
-	if err := app.Run(os.Args); err != nil {
-		fmt.Println(err)
-	}
-}
-
-func generateFishCompletion(c *cli.Context) error {
-	fishScript, err := c.App.ToFishCompletion()
-	if err != nil {
-		return err
-	}
-
-	outPath := c.String("out")
-	if outPath != "" {
-		if err := os.WriteFile(outPath, []byte(fishScript), 0o644); err != nil {
-			return err
-		}
-	} else {
-		fmt.Println(fishScript)
-	}
-
-	return nil
+	fmt.Fprintln(os.Stderr, `The "livekit-cli" binary has been renamed to "lk". Try again with the new name!`)
```

Comment:
I wonder, do we really have to do this? Can we symlink instead? CLI could also check its own binary name (aka `os.Args[0]`) and print a migration warning.

Since we will hide all CLI commands and they will continue to work anyway, there's no real breaking change here. Why are we specifically breaking the CLI under the old name?

Reply:
I think, because we can with a 2.0 release? Yes we support the old commands, but we don't want to actively encourage their use and they will not be maintained.

I agree a symlink and detecting $0 is more elegant than a separate (~1.4MB) binary, but it's also more fiddly. Not that it's hard, but a simple solution is also simple to remove when we feel ready.

Reply:
> I think, because we can with a 2.0 release?

Right, if you proposed to remove old CLI _and_ old sub-commands for v2 that would be understandable. But we are keeping old commands still. So why not keep the binary as well (as a symlink at least)?

> but it's also more fiddly

Definitely more fiddly for us, but the users might appreciate it :man_shrugging:

Reply:
+1 to including a warning, but don't break immediately

Reply:
OK, removed the dummy binary and added symlinking. The CLI will now print the following warning if `os.Args[0]` does not end with "lk":

```
tobiasfried@slate bin % livekit-cli --help

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ DEPRECATION NOTICE ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The `livekit-cli` binary has been renamed to `lk`, and some of the options and
commands have changed. Though legacy commands my continue to work, they have
been hidden from the USAGE notes and may be removed in future releases.
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

NAME:
   lk - CLI client to LiveKit

USAGE:
   lk [global options] [command [command options]] [arguments...]

VERSION:
   2.0.0

DESCRIPTION:
   A suite of command line utilities allowing you to access LiveKit APIs services, interact with rooms in realtime, and perform load testing simulations.

COMMANDS:
   Core:
     token    Create access tokens with granular capabilities
     room     Create or delete rooms and manage existing room properties
     project  Add or remove projects and view existing project properties
   I/O:
     egress   Record or stream media from LiveKit to elsewhere
     ingress  Import outside media sources into a LiveKit room
     sip      Manage SIP Trunks, Dispatch Rules, and Participants
   Simulate:
     load-test  Run load tests against LiveKit with simulated publishers & subscribers

GLOBAL OPTIONS:
   --url URL            URL to LiveKit instance (default: "http://localhost:7880") [$LIVEKIT_URL]
   --api-key KEY        Your KEY [$LIVEKIT_API_KEY]
   --api-secret SECRET  Your SECRET [$LIVEKIT_API_SECRET]
   --project NAME       NAME of a configured project
   --curl               Print curl commands for API actions (default: false)
   --verbose            (default: false)
   --help, -h           show help (default: false)
   --version, -v        print the version (default: false)
```